---
title: "R Notebook"
output: html_notebook
---
```{r message=F}
library(tidyverse, quietly = T)
library(readr, quietly = T)
```

```{r}
# Assign column names according to the GENCODE instruction.
cols = c('chrom', 'source', 'feature_type', 'start', 'end', 'score', 'strand', 'phase', 'info')
```

```{r}
d = read_delim('gencode.v31.basic.annotation.gtf.gz', 
               delim='\t', skip = 5, 
               progress = F,
               col_names = cols)
```

```{r}
d2 <- d %>% filter(feature_type == 'transcript')
```

```{r}
## If the data is too big, use this method
write.table(d2, 'table.transcript_genecode.20190923.txt', sep='\t', quote=F, row.names = F, col.names = T)
```

```{r}
d=read_delim('table.transcript_genecode.20190923.txt',delim='\t', skip = 5, 
               progress = F,
               col_names = cols)
```



```{r}
## transcript_support_level
d$transcript_support_level <- as.character(do.call(rbind.data.frame, 
                                                    strsplit(d$info, 'transcript_support_level\\s+"'))[[2]])

d$transcript_support_level <- as.character(do.call(rbind.data.frame, 
                                                    strsplit( d$transcript_support_level, '\\"'))[[1]])
```

```{r}
## gene_type
d$gene_type <- as.character(do.call(rbind.data.frame, 
                                                    strsplit(d$info, 'gene_type\\s+"'))[[2]])

d$gene_type <- as.character(do.call(rbind.data.frame, 
                                                    strsplit( d$gene_type, '\\"'))[[1]])
```

1. With `transcript`, how many transcripts are categorized for each TSL?
```{r}
## Be aware of 'space bar'
d$transcript_support_level<-ifelse(d$transcript_support_level=="gene_id ", "NA", d$transcript_support_level)
```

```{r}
d %>% group_by(transcript_support_level) %>%
  summarise(total = n())
```


2. From the first question, please count the number of transcript for each TSL by gene biotype.
```{r}
d %>% group_by(transcript_support_level,gene_type) %>%
  summarise(total = n())
```


3. From the first question, please count the number of transcript for each TSL by source.
```{r}
d %>% group_by(transcript_support_level,source) %>%
  summarise(total = n())
```


