---
title: "R Notebook"
output: html_notebook
---
```{r message=F}
library(tidyverse, quietly = T)
library(readr, quietly = T)
```

```{r}
d = read_delim('gencode.v31.basic.annotation.gtf.gz', 
               delim='\t', skip = 5, progress = F, 
               col_names = F)
```

```{r}
# Assign column names according to the GENCODE instruction.
cols = c('chrom', 'source', 'feature_type', 'start', 'end', 'score', 'strand', 'phase', 'info')
```

```{r}
d = read_delim('gencode.v31.basic.annotation.gtf.gz', 
               delim='\t', skip = 5, 
               progress = F,
               col_names = cols)
```

```{r}
# First filter transcripts and create a data frame.
d2 <- d %>% filter(feature_type == 'transcript')

# Now apply the functions. 
d2$transcript_support_level <- as.character(do.call(rbind.data.frame, 
                                                    strsplit(d2$info, 'transcript_support_level\\s+"'))[[2]])

d2$transcript_support_level <- as.character(do.call(rbind.data.frame, 
                                                    strsplit( d2$transcript_support_level, '\\"'))[[1]])
```

```{r}
# Now apply the functions. 
d2$gene_type <- as.character(do.call(rbind.data.frame, 
                                                    strsplit(d2$info, 'gene_type\\s+"'))[[2]])

d2$gene_type <- as.character(do.call(rbind.data.frame, 
                                                    strsplit( d2$gene_type, '\\"'))[[1]])
```

```{r}
TSL<-data_frame(transcript_support_level = d2$transcript_support_level,
                gene_type = d2$gene_type,
                source = d$source)
```


1. With `transcript`, how many transcripts are categorized for each TSL?
```{r}
table(TSL$transcript_support_level)
```


2. From the first question, please count the number of transcript for each TSL by gene biotype.
```{r}
table(TSL$gene_type)
```


3. From the first question, please count the number of transcript for each TSL by source.
```{r}
table(TSL$source)
```


